### PR TITLE
gimp: fix all plugins build

### PR DESCRIPTION
**Pull request by avnik**
###### Motivation for this change

Some plugins was broken, and prevent build gimp with all plugins installed

###### Things done

- [x] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/graphics/gimp/plugins/default.nix [hunk 1]
```diff
@@ -125,10 +127,11 @@ rec {
...
     */
     name = "resynthesizer-2.0.1";
-    buildInputs = [ gimp pkgs.fftw pkgs.autoreconfHook pkgs.pkgconfig pkgs.gtk2
-      pkgs.intltool
-    ]
-      ++ gimp.nativeBuildInputs;
+    buildInputs = with pkgs; [ 
+      fftw 
+      autoreconfHook
+      intltool
```
Comment (Mic92) [inline pkgs/applications/graphics/gimp/plugins/default.nix]: autoreconfHook and intltool belong to `nativeBuildInputs`

File: pkgs/applications/graphics/gimp/plugins/default.nix [hunk 2]
```diff
@@ -140,11 +143,11 @@ rec {
 
   texturize = pluginDerivation {
     name = "texturize-2.1";
-    buildInputs = [ gimp ] ++ gimp.nativeBuildInputs;
     src = fetchurl {
       url = mirror://sourceforge/gimp-texturize/texturize-2.1_src.tgz;
       sha256 = "0cdjq25g3yfxx6bzx6nid21kq659s1vl9id4wxyjs2dhcv229cg3";
     };
+    buildInputs = with pkgs; [ intltool perl ];
```
Comment (Mic92) [inline pkgs/applications/graphics/gimp/plugins/default.nix]: intltool belongs to nativeBuildInputs

File: pkgs/applications/graphics/gimp/plugins/default.nix [hunk 3]
```diff
@@ -156,7 +159,7 @@ rec {
       Filters/Enhance/Wavelet sharpen
     */
     name = "wavelet-sharpen-0.1.2";
-    buildInputs = [ gimp ] ++ gimp.nativeBuildInputs;
+    buildInputs = with pkgs; [ intltool ];
```
Comment (Mic92) [inline pkgs/applications/graphics/gimp/plugins/default.nix]: same here.

File: pkgs/applications/graphics/gimp/plugins/default.nix [hunk 4]
```diff
@@ -169,7 +172,7 @@ rec {
        Layer/Liquid Rescale
     */
     name = "lqr-plugin-0.6.1";
-    buildInputs = [ pkgconfig libLQR gimp ] ++ gimp.nativeBuildInputs;
+    buildInputs = with pkgs; [ intltool libLQR ];
```
Comment (Mic92) [inline pkgs/applications/graphics/gimp/plugins/default.nix]: same here.

File: pkgs/applications/graphics/gimp/plugins/default.nix [hunk 5]
```diff
@@ -23,7 +23,10 @@ let
   }
   // a
     # don't call this gimp-* unless you want nix replace gimp by a plugin :-)
-  // { name = "${a.name}-${gimp.name}-plugin"; }
+  // {
+      name = "${a.name}-${gimp.name}-plugin";
+      buildInputs = [ gimp gimp.gtk glib pkgconfig ] ++ (a.buildInputs or []);
```
Comment (Mic92) [inline pkgs/applications/graphics/gimp/plugins/default.nix]: `pkgconfig` belongs to nativeBuildInputs.

**Review by Mic92 — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/gimp/plugins/default.nix (hunk 1 above)
Comment (avnik) [inline pkgs/applications/graphics/gimp/plugins/default.nix]: intltool is enough common for most plugins, should it be default?

File: pkgs/applications/graphics/gimp/plugins/default.nix (hunk 5 above)
Comment (avnik) [inline pkgs/applications/graphics/gimp/plugins/default.nix]: Thinked about split  nativeBuildInputs, but was urge to rebuild my system. Will do.

**Comment by avnik** [comment]
@Mic92  all stuff you requested was fixed

File: pkgs/applications/graphics/gimp/plugins/default.nix (hunk 1 above)
Comment (Mic92) [inline pkgs/applications/graphics/gimp/plugins/default.nix]: could be.

**Comment by avnik** [comment]
@Mic92  also promoted intltool as default build input,  I think it ready to merge

**Pull request merged**